lint/rule_target_rate_interval.go

```diff
@@ -0,0 +1,85 @@
+package lint
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+type inspector func(parser.Node, []parser.Node) error
+
+func (f inspector) Visit(node parser.Node, path []parser.Node) (parser.Visitor, error) {
+	if err := f(node, path); err != nil {
+		return nil, err
+	}
+	return f, nil
+}
+
+// NewTargetRateIntervalRule builds a lint rule for panels with Prometheus queries which checks
+// all range vector selectors use $__rate_interval.
+func NewTargetRateIntervalRule() *TargetRuleFunc {
+	rateIntervalMagicDuration, err := time.ParseDuration(globalVariables["__rate_interval"].(string))
+	if err != nil {
+		// Will not happen
+		panic(err)
+	}
+	return &TargetRuleFunc{
+		name:        "target-rate-interval-rule",
+		description: "Checks that each target uses $__rate_interval.",
+		fn: func(d Dashboard, p Panel, t Target) Result {
+			if t := getTemplateDatasource(d); t == nil || t.Query != Prometheus {
+				// Missing template datasources is a separate rule.
+				return ResultSuccess
+			}
+
+			if !panelHasQueries(p) {
+				// Don't lint certain types of panels.
+				return ResultSuccess
+			}
```

Comment:
I'm not convinced this check is necessary anymore, since this will *only* be run for panels which have targets (so it would naturally exclude rows, text, and other non dataviz panels) and we're already checking that the panel is of the correct "type", namely that it uses a prometheus DS.